### PR TITLE
[pigeon] Made using enums in type arguments an error, added workaround

### PR DESCRIPTION
**Pull request by gaaclarke**
There are a lot of features related to enum's we don't yet support.  Issuing an error in the front-end is better than a compilation error later.

As a workaround I started always generating enums in code whether they were referenced or not so users can use their integer values.

issues: 
 * https://github.com/flutter/flutter/issues/93525
 * https://github.com/flutter/flutter/issues/93464

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/pigeon/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
-## NEXT
+## 1.0.11
 
 * [ci] Started transitioning to a Dart test runner, added windows support.
+* [front-end] Started issuing an error if enums are used in type arguments.
+* [front-end] Passed through all enums, referenced or not so they can be used as
+  a work around for direct enum support.
```
Comment (stuartmorgan) [inline packages/pigeon/CHANGELOG.md:6]: Nit: There's now [a style guide for CHANGELOGs](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style), so these should follow it (e.g., Starts issuing, Passes through)

File: packages/pigeon/lib/pigeon_lib.dart [hunk 2]
```diff
@@ -544,9 +551,6 @@ class _RootBuilder extends dart_ast_visitor.RecursiveAstVisitor<Object?> {
         .removeWhere((Class x) => !referencedTypeNames.contains(x.name));
 
     final List<Enum> referencedEnums = List<Enum>.from(_enums);
-    referencedEnums.removeWhere(
```
Comment (stuartmorgan) [inline packages/pigeon/lib/pigeon_lib.dart:547]: Do you want to leave a TODO here to re-add this?

**Review by stuartmorgan — APPROVED** [review]
LGTM with nits

File: packages/pigeon/CHANGELOG.md (hunk 1 above)
Comment (gaaclarke) [reply packages/pigeon/CHANGELOG.md:6]: Done

File: packages/pigeon/lib/pigeon_lib.dart (hunk 2 above)
Comment (gaaclarke) [reply packages/pigeon/lib/pigeon_lib.dart:547]: Not necessarily.  Only generating output for referenced types has lead to a lot of bugs.  It was necessary when we we were using reflection, now it isn't needed.  If we start supporting multiple imports in pigeon files we might have to revisit it again.

**Pull request merged**